e2e/common/world.ts

```diff
@@ -12,6 +18,33 @@ export const expect = chai.expect;
 // we need to export Promise as P because 'Promise' is reserved for async/await usage
 export { P };
 
+export const OAUTH_TOKEN_NAME = 'access_token';
+
+/**
+ * Wrap http status code and response data into single object.
+ */
+export interface IpaasResult {
+  statusCode: number;
+  data: any;
+}
+
+/**
+ * Helper for easier wrapping of callbacks into Promise.
+ */
+export class Deferred<T> {
+
+  promise: P<T>;
+  resolve: (value?: T | PromiseLike<T>) => void;
+  reject: (reason?: any) => void;
+
+  constructor() {
+    this.promise = new P<T>((resolve, reject) => {
+      this.resolve = resolve;
+      this.reject = reject;
+    });
+  }
+}
```

Comment:
to allow 
```ts
const deferred: Deferred<IpaasResult> = new Deferred();
deferred.resolve();
return deferred.promise;
```

instead of nesting whole block into 
```ts
return new Promise((resolve, reject) => {
 // into here
});
```